### PR TITLE
add helper method to build CronetEngine

### PR DESCRIPTION
**Pull request by mnbogner**
calls to build a CronetEngine have been moved into a helper method to ensure consistency

### CODE REVIEW

File: android/envoy/src/main/kotlin/org/greatfire/envoy/CronetNetworking.kt [hunk 1]
```diff
@@ -27,50 +27,42 @@ object CronetNetworking {
...
+            Log.d("FOO", "FOUND CACHE SETTINGS FOR BUILDER: " + cacheFolder)
+            val cacheDir = File(context.cacheDir, cacheFolder)
+            if (!cacheDir.exists()) {
+                cacheDir.mkdirs()
+            }
+            builder = builder
+                .setStoragePath(cacheDir.absolutePath)
+                .enableHttpCache(CronetEngine.Builder.HTTP_CACHE_DISK, cacheSize * 1024 * 1024)
+        } else {
+            Log.d("FOO", "FOUND NO CACHE SETTINGS FOR BUILDER")
```
Comment (vitriolix) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/CronetNetworking.kt:51]: maybe clean up the FOO logs

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt [hunk 2]
```diff
@@ -904,25 +899,20 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
         Log.d(TAG, "create request to " + captive_portal_url + " for url: " + UrlUtil.sanitizeUrl(envoyUrl, envoyService))
 
         val sanitizedOriginal = UrlUtil.sanitizeUrl(originalUrl, envoyService)
+        val cacheFolder = cacheMap.get(originalUrl)
 
-        if (cacheMap.keys.contains(originalUrl)) {
+        if (!cacheFolder.isNullOrEmpty()) {
```
Comment (vitriolix) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:904]: I'm not sure i see how this is related to this pr?

File: android/envoy/src/main/kotlin/org/greatfire/envoy/CronetNetworking.kt (hunk 1 above)
Comment (mnbogner) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/CronetNetworking.kt:51]: definitely, i just left them in to do a test run

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt (hunk 2 above)
Comment (mnbogner) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:904]: i think the previous version of the helper had a non-nullable cacheFolder argument.  i think i can revert this.

**Pull request merged**